app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -183,53 +184,52 @@ export default class TransactionStateManager extends EventEmitter {
     const snapshot = snapshotFromTxMeta(txMeta);
     txMeta.history.push(snapshot);
 
-    const transactions = this.getFullTxList();
+    const transactions = this.getTransactions();
```

Comment:
this needs to specify { filterToCurrentNetwork: false } 😬 to preserve the previous behavior. This change results in a functional change which is that 40 transactions *per network* would be saved.